### ISSUE TITLE
Support xdg-open

### ISSUE THREAD

**Issue by jacobembree** [ralphembree/mintmenu-helper#2]
xdg-open does the same thing as gnome-open.  Mint 18 has the former not the latter.  We should use whichever is available.
I don't mind doing it at some point when I have time.
Ways I can think of for checking availability of a program:

- `which gnome-open` has output
- `! hash gnome-open 2> /dev/null`
I forget how this works.  I think `hash` is a bash builtin.  If it is we shouldn't use it in case bash isn't available.

Hopefully you, @ralphembree, know a Python way.

**Comment by jacobembree** [comment]
We should also support `gvfs-open`.  See https://askubuntu.com/questions/101965/what-is-the-replacement-for-gnome-open-in-gnome.

**Comment by ralphembree** [comment]
xdg-open seems to be the most commonly used one, so I would expect that to work in almost every case. If that isn't installed, a FileNotFoundError will be raised. I would then try gnome-open first and then gvfs-open. Python has `os.startfile` for opening something with the default application, but it is available only on Windows.

**Comment by ralphembree** [comment]
I found a StackOverflow answer that suggests webbrowser.open() (https://stackoverflow.com/a/11479099/5827958). I tested it on a directory, PNG, PDF, plain text, HTML, executable, MP3, and .tar.gz. Despite the module in which it is defined, it seemed to have no bias towards opening things in my browser, but opened everything in the default application. I then found the source code for webbrowser (https://github.com/python/cpython/blob/a7cba27aea138311117e2ab1d91584efcfeac4ec/Lib/webbrowser.py#L438) which shows that xdg-open is first in line, then gvfs-open, then gnome-open. It tries to figure it out itself if all those work, so this seems to be the best solution.
